.devops/azure-templates/deploy-client-sdk.yml

```
@@ -0,0 +1,71 @@
+# Azure DevOps pipeline template to release Client SDK to NPM
+
+parameters:
+
+  - name: 'dependsOn'
+    type: string
+    default: 'No_Name'
+
+  - name: 'projectDir'
+    type: string
+    default: '.'
+
+  - name: 'openapiSpecPath'
+    type: string
+    default: 'openapi/index.yaml'
+
+
+stages:
+
+ # Prepare SDK
+ # make this stage run after Release, so package.json will be automatically updated
+ - stage: Prepare_SDK_Artifact_And_Release  
+   dependsOn: ['${{parameters.dependsOn}}']
+   
+
+   variables:
+    openapiSpecPath_withProjectDir: '${{ parameters.projectDir }}/${{ parameters.openapiSpecPath }}'
+    generatedCodeDir: '${{ parameters.projectDir }}/generated/sdk'
```

Maybe we can try `Agent.TempDirectory` for generated files? https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#agent-variables-devops-services

```suggestion
    generatedCodeDir: '$(Agent.TempDirectory)/generated'
```